### PR TITLE
feature 394 search resumes by skill

### PR DESCRIPTION
**Pull request by alex-flores-jalafund**
[AB#378](https://dev.azure.com/funjaladev/DevLevel3/_boards/board/t/DevLevel3%20Team/Stories/?workitem=378)

# Overview

As a project user I want to search resumes by skills
image

### CODE REVIEW

File: src/dal/Jalasoft.TeamUp.Resumes.DAL/ResumesInMemoryRepository.cs [hunk 1]
```diff
@@ -150,6 +150,11 @@ public IEnumerable<Skill> UpdateResumeSkill(int idResume, Skill[] skills)
         public IEnumerable<Resume> GetByName(string name)
         {
             return Resumes;
-        }
+        }
+
+        public IEnumerable<Resume> GetBySkill(string skill)
+        {
+            throw new NotImplementedException();
+        }
```
Comment (alejandro-castellon) [inline src/dal/Jalasoft.TeamUp.Resumes.DAL/ResumesInMemoryRepository.cs]: There is no inmemory function

File: src/dal/Jalasoft.TeamUp.Resumes.DAL.Interfaces/IRepository.cs [hunk 2]
```diff
@@ -16,6 +16,8 @@ public interface IRepository<T>
 
         public void Delete(int id);
 
-        public IEnumerable<Resume> GetByName(string name);
+        public IEnumerable<Resume> GetByName(string name);
+
+        public IEnumerable<Resume> GetBySkill(string skill);
```
Comment (alejandro-castellon) [inline src/dal/Jalasoft.TeamUp.Resumes.DAL.Interfaces/IRepository.cs]: This should go to IResumesRepository

File: src/api/Jalasoft.TeamUp.Resumes.API/Controllers/GetResumesBySkill.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            this.getResumeBySkill = skillsService;
+        }
+
+        [FunctionName("GetResumeBySkill")]
+        [OpenApiOperation(operationId: "GetResumesBySkill", tags: new[] { "Resumes" })]
+        [OpenApiParameter(name: "skill", In = ParameterLocation.Query, Required = true, Type = typeof(string), Description = "The name of the skill to search by.")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Resume[]), Description = "Successful response")]
+        [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.NotFound, Description = "Resource not found")]
+        public IActionResult Run(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "v1/getResumesBySkill")] HttpRequest req)
```
Comment (alejandro-castellon) [inline src/api/Jalasoft.TeamUp.Resumes.API/Controllers/GetResumesBySkill.cs]: Maybe "v1/{skill}/resumes" instead of that route

**Review by alejandro-castellon — CHANGES_REQUESTED** [review]

**Review by ela-ocana-jalafund — DISMISSED** [review]

File: src/api/Jalasoft.TeamUp.Resumes.API/Controllers/GetResumes.cs [hunk 4]
```diff
@@ -21,24 +22,28 @@ public GetResumes(IResumesService resumesService)
...
             }
-            catch (ResumesException e)
+            catch (ResumesException ex)
             {
-                return e.Error;
+                var error = ex.Error;
+                return error;
             }
             catch (System.Exception e)
             {
```
Comment (alejandro-castellon) [inline src/api/Jalasoft.TeamUp.Resumes.API/Controllers/GetResumes.cs:49]: Are you overwriting the getResumes service?

**Comment by alejandro-castellon** [comment]
And did you add inmemory part?

**Review by alejandro-castellon — DISMISSED** [review]
Approve gg

File: src/api/Jalasoft.TeamUp.Resumes.API/Controllers/GetResumes.cs (hunk 4 above)
Comment (alex-flores-jalafund) [reply src/api/Jalasoft.TeamUp.Resumes.API/Controllers/GetResumes.cs:49]: it's for an endpoint

**Review by ela-ocana-jalafund — DISMISSED** [review]

**Review by alejandro-castellon — DISMISSED** [review]

File: tests/core/Jalasoft.TeamUp.Resumes.Core.Tests/GetResumesServiceTests.cs [hunk 5]
```diff
@@ -147,6 +147,44 @@ public void GetResume_UnexistentID_ReturnNull()
...
+            // Act
+            var result = this.resumesService.GetResumes("C#");
+
+            // Assert
+            Assert.Equal(2, result.Length);
+        }
+
+        [Fact]
+        public void GetResumesBySkill_NotExist_Skill()
+        {
```
Comment (Raul-Gamarra-jalafund) [inline tests/core/Jalasoft.TeamUp.Resumes.Core.Tests/GetResumesServiceTests.cs]: this unit test is not needed

**Review by ela-ocana-jalafund — APPROVED** [review]

File: tests/core/Jalasoft.TeamUp.Resumes.Core.Tests/GetResumesServiceTests.cs (hunk 5 above)
Comment (alex-flores-jalafund) [reply tests/core/Jalasoft.TeamUp.Resumes.Core.Tests/GetResumesServiceTests.cs]: it's ok

**Review by alejandro-castellon — APPROVED** [review]

**Review by Raul-Gamarra-jalafund — APPROVED** [review]

**Pull request merged**